task-tracker/src/app/task-list/task-list.component.ts

```diff
@@ -32,61 +38,90 @@ export class TaskListComponent implements OnInit {
     this.updatePagination();
   }
 
-  onTaskAdded(newTask: any) {
```

Comment:
dla metody filterTasks:
if(this.searchQuery.trim() !== '') {
this.tasks = this.tasks.filter(task => 
      task.description.toLowerCase().includes(this.searchQuery.toLowerCase())
    );
this.paginateTasks()
} else {
this.loadTasks();
}